src/graphics/shader.rs

```diff
@@ -231,23 +231,89 @@ impl<'a, Uniforms: AsStd140> ShaderParamsBuilder<'a, Uniforms> {
     }
 
     /// Produce a [ShaderParams] from the builder.
-    pub fn build(self, gfx: &mut impl HasMut<GraphicsContext>) -> ShaderParams<Uniforms> {
-        let gfx = gfx.retrieve_mut();
-        let uniforms = ArcBuffer::new(gfx.wgpu.device.create_buffer_init(
-            &wgpu::util::BufferInitDescriptor {
-                label: None,
-                usage: wgpu::BufferUsages::UNIFORM | wgpu::BufferUsages::COPY_DST,
-                contents: self.uniforms.as_std140().as_bytes(),
-            },
-        ));
+    pub fn build(self, ctx: &mut Context) -> ShaderParams<Uniforms> {
```

Comment:
should've written the comment in the review

Reply:
ctx.time is used in set_uniforms (which is called by build)